services/app-graphql/src/schema.graphql

```diff
@@ -1194,8 +1247,106 @@ type UnlockRatePayload {
+    or an amendment to an existing rate certification
+    """
+    rateType: RateType
+    """
+    Can be 'RATE_CELL' or 'RATE_RANGE'
+    These values represent on what basis the capitation rate is actuarially sound
+    """
+    rateCapitationType: RateCapitationType
+    """
+    Signed certification documents the state uploads
+    Files can be PDF, DOC, or DOCX format
+    """
+    rateDocuments: [GenericDocumentInput!]!
+    """
+    Additional documents the state uploads to support a rate cert
+    Files can be PDF, DOC, DOCX, XLSX, CSV format
+    """
+    supportingDocuments: [GenericDocumentInput!]!
+    """
+    If the rateType is NEW this is the start date of the
+    rating period for a new certification.
+    If the rateType is AMENDMENT this is the start date of the
+    rating period for the original rate certification
+    """
+    rateDateStart: Date
+    """
+    If the rateType is NEW this is the end date of the
+    rating period for a new certification.
+    If the rateType is AMENDMENT this is the end date of the
+    rating period for the original rate certification
+    """
+    rateDateEnd: Date
+    """
+    The date the rate certification was
+    certified/signed by the state's actuary
+    """
+    rateDateCertified: Date
+    """
+    The start date of the rate amendment
+    Only relevant if rate type is AMENDMENT
+    """
+    amendmentEffectiveDateStart: Date
+    """
+    The end date of the rate amendment
+    Only relevant if rate type is AMENDMENT
+    """
+    amendmentEffectiveDateEnd:  Date
+    """
+    An array of IDs representing state programs that the rate covers
+    """
+    rateProgramIDs: [String!]!
+    """
+    Represents the name of the rate.
+    This value is auto generated based on rate, package and state program details
+    """
+    rateCertificationName: String
+    """
+    An array of ActuaryContacts
+    Each element includes the the name, title/role and email
+    of the actuaries who certified the rate
+    """
+    certifyingActuaryContacts: [ActuaryContactInput!]!
+    """
+    An array of additional ActuaryContacts
+    Each element includes the the name, title/role and email
+    """
+    addtlActuaryContacts: [ActuaryContactInput!]!
+    """
+    Is either OACT_TO_ACTUARY or OACT_TO_STATE
+    It specifies whether the state wants CMS to reach out to their actuaries
+    directly or go through them
+    """
+    actuaryCommunicationPreference: ActuaryCommunication
+    """
+    An array of PackageWithSameRate elements
+    which contain the packageName, packageId, and packageStatus
+    These elements represent other packages in the system
+    that are using this rate
+    """
+    packagesWithSharedRateCerts: [PackageWithSameRateInput!]!
+}
+
+input SubmitRateInput {
+    rateID: ID!
+    "User given submission description"
+    submitReason: String!
+    "Rate related form data to be updated with submission"
+    formData: RateFormDataInput!
```

Comment:
🛑 Could we  just use `formData: RateFormData` and use the existing type here? Is there a difference between these types?

 I would really like compose together existing types rather than duplicate types between the `type` and the `input` when possible.

Reply:
I tried, but it gave errors. From my understanding, GraphQL `input` and `types` are different, and only `input` can be used as arguments for a mutation. I think this [StackOverflow](https://stackoverflow.com/questions/41743253/whats-the-point-of-input-type-in-graphql/55881719#55881719) answer gives a better explanation than I could. It's an old thread, but I think it is still correct.

Reply:
Yes okay, I was looking at that too. Alright ✅